contracts/contracts/vault/VaultCore.sol

```diff
@@ -385,32 +385,102 @@ contract VaultCore is VaultInitializer {
      * @return totalUnits Total balance of Vault in units
      */
     function _rebase() internal whenNotRebasePaused returns (uint256) {
-        uint256 ousdSupply = oUSD.totalSupply();
+        uint256 supply = oUSD.totalSupply();
         uint256 vaultValue = _totalValue();
-        if (ousdSupply == 0) {
+        uint256 nonRebasing = oUSD.nonRebasingSupply();
+        // If no supply yet, do not rebase
+        if (supply == 0) {
             return vaultValue;
         }
 
-        // Yield fee collection
+        // Calculate yield and new supply
+        uint256 yield = 0;
+        uint256 newTarget = 0;
+        (yield, newTarget) = _nextYield(supply, vaultValue, nonRebasing);
+        uint256 newSupply = supply + yield;
+        // Only rebase upwards and if we have enough backing funds
+        if (newSupply <= supply || newSupply > vaultValue) {
```

Comment:
The `newSupply > vaultValue` is probably a sanity check? Though from the current code I don't see how this condition could ever be hit

Reply:
Yes. Just an extra layer of sanity check, since it means that if you do something really dumb or have a bug in _nextYield, then you shouldn't be able to rekt the protocol.